docs/contributing/inheritance.md

```diff
@@ -0,0 +1,69 @@
+# Stravalib's Inheritance Patterns
+
+Stravalib's API is built directly from Strava's `swagger.json` example response
+data. We use the Pydantic `BaseModel` combined with `datacodegen` to build our
+base model objects (found in `strava_model.py`), which we then enhance and
+update to align with the data actually returned by the Strava API.
+
+This page will help you understand Stravalib's inheritance patterns.
+
+In our experience maintaining Stravalib, the Strava API online specification
+and the `swagger.json` file don't always align perfectly with the data actually
+returned by Strava. As a result, you will notice that we frequently overwrite
+data types to match the actual data we see returned from the Strava API for
+each endpoint. [More information on this process can be found here.](ci_api_updates)
+
+Below, we present the inheritance schema for Stravalib.
+
+## Inheritance overview
+
+At a high level, there are two modules:
+
+* {py:mod}`stravalib.model` and
+* {py:mod}`stravalib.strava_model`.
+
+`strava_model` is generated automatically using a CI build from the Strava API.
+Stravalib uses the model.py module to do a few things:
+
+1. It supports inheritance of the `BoundClientEntity`, which supports API calls for lazily loaded properties,
+2. it allows us to override `strava_model` attributes that have typed attributes that don't align with the actual API responses, and
+3. allows us to add attributes that are found in the returned data but not documented or found in the swagger.json response.
+
+The full inheritance pattern, that includes inheritance from both `strava_model` and `pydantic.BaseModel` is below.
+
+:::{inheritance-diagram} model.DetailedAthlete
```

Comment:
@jsamoocha i'm fighting a bit with the diagrams - but mermaid seems to be nice. The catch so far has been that i can't seem to escape periods. How does this look to you as a way to show the relationship between model and strava_model at a high level? I added labels to make this work for the model module. 

<img width="642" alt="Screenshot 2024-09-15 at 5 21 41 PM" src="https://github.com/user-attachments/assets/a710a04e-54e0-49e0-b3c1-b1bcd231ac0f">